### PR TITLE
Initial commit for package libnetworkit

### PR DESCRIPTION
**Pull request by fabratu**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by fabratu** [comment]
As of now this C++ library is part of https://github.com/conda-forge/networkit-feedstock. The plan is to split Python/C++ parts. 

Once/If this package is accepted an update will be uploaded on the related package, adding a dependency to this package and removing the library there.

**Comment by avdgrinten** [comment]
I confirm that I will act as a maintainer.

**Comment by scopatz** [comment]
Thanks for putting this in.  Please resolve the linter bot issues :wink:

**Comment by fabratu** [comment]
@scopatz Done :smirk:

File: recipes/libnetworkit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+source:
+  - url: 'https://github.com/networkit/networkit/archive/{{version}}.tar.gz'
+    {{hash_type}}: {{hash_value}}
+  - git_url: 'https://github.com/google/googletest.git'
+    git_tag: '2fe3bd994b3189899d93f1d5a881e725e046fdc2'
+    folder: 'extlibs/googletest/'
+  - git_url: 'https://github.com/tlx/tlx'
+    git_tag: 'd5011fa9a1e0cb6584ac34e3e84550e5d45bda1a'
+    folder: 'extlibs/tlx/'
```
Comment (scopatz) [inline recipes/libnetworkit/meta.yaml]: Can you use the gtest package? Also, is there a tlx package that you can use?

**Comment by scopatz** [comment]
This seems to vendor other packages still, so that needs to be resolved

**Comment by fabratu** [comment]
Sry for the delay (and happy new year!). The external package "tlx" is currently in a PR-state: #10542 

Once this is approved, I will update this package.

**Comment by angriman** [comment]
I confirm that I will act as a maintainer.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**